daft/execution/physical_plan.py

```diff
@@ -38,19 +38,22 @@
 )
 from daft.expressions import ExpressionsProjection
 from daft.logical.schema import Schema
-from daft.runners.partitioning import PartialPartitionMetadata
+from daft.runners.partitioning import (
+    MaterializedResult,
+    PartialPartitionMetadata,
+    PartitionT,
+)
 
 logger = logging.getLogger(__name__)
 
-PartitionT = TypeVar("PartitionT")
 T = TypeVar("T")
 
 
 # A PhysicalPlan that is still being built - may yield both PartitionTaskBuilders and PartitionTasks.
 InProgressPhysicalPlan = Iterator[Union[None, PartitionTask[PartitionT], PartitionTaskBuilder[PartitionT]]]
 
 # A PhysicalPlan that is complete and will only yield PartitionTasks or final PartitionTs.
-MaterializedPhysicalPlan = Iterator[Union[None, PartitionTask[PartitionT], PartitionT]]
+MaterializedPhysicalPlan = Iterator[Union[None, PartitionTask[PartitionT], MaterializedResult[PartitionT]]]
```

Comment:
NOTE this important change: instead of yielding `PartitionT` (either `Table` or `ray.ObjectRef`), we now yield the `MaterializedResult` container instead.